engine/json.go

```diff
@@ -54,6 +54,18 @@ type RawMiddleware struct {
 	Middleware json.RawMessage
 }
 
+type rawListener struct {
+	Id       string
+	Protocol string
+	Address  Address
+	Scope    string
+	Settings json.RawMessage
```

Comment:
If you embed `HTTPSListenerSettings` directly here you could remove `HTTPSListenerSettingsFromJSON` and `TLSSettingsFromJSON` below.

https://gist.github.com/jeremyschlatter/64cddb7597c8a85059ae